gwpy/table/io/losc.py

```diff
@@ -102,17 +101,22 @@ def _mask_column(col):
 @read_with_columns
 @read_with_selection
 def fetch_catalog(catalog, host=DEFAULT_GWOSC_URL):
-    catalog = download_catalog(catalog, host=host)
-    data = catalog["data"]
-    parameters = catalog["parameters"]
+    catalog = fetch_catalog_json(catalog, host=host)
+    data = catalog["events"]
+
+    firstdata = next(iter(data))
+    parameters = list(data[firstdata])
```

Comment:
I'm afraid that this might encode an incorrect assumption about the GWOSC catalogs, namely that all events in the catalog have the same parameters.

@jkanner, can you confirm whether all events in the eventapi catalog views are guaranteed to have the same parameters? Or should we prepare for the eventually that some events have data that others don't? I'm imagining mixed catalogs of CBC and burst events in the future.

@rngeorge, the safest approach would be to take the logical OR of all available parameters:

```python
parameters = set(key for event in data.values() for key in event)
```

and then reinstate the 'masking' we had before to handle rows that have missing parameters.

Reply:
Hi Duncan,

That's interesting.  For now, yes, all events have the same parameters at the top level.  Though, individual PE sets may have an arbitrary list of parameters (and each PE set may be different). 

We haven't yet thought about how to represent different types of events (like Supernova or pulsars) - for now, they are all CBC.  I agree that thinking ahead, and imagining a heterogenous list at some point in the future is good planning.